crates/chain/src/keychain/txout_index.rs

```diff
@@ -5,11 +5,85 @@ use crate::{
     spk_iter::BIP32_MAX_INDEX,
     SpkIterator, SpkTxOutIndex,
 };
+use alloc::string::{String, ToString};
 use bitcoin::{OutPoint, Script, TxOut};
 use core::{fmt::Debug, ops::Deref};
+use miniscript::descriptor::checksum::desc_checksum;
 
 use crate::Append;
 
+type DescriptorId = String;
```

Comment:
String is not nice as it is variable-length.

Additionally, we can use a byte array (instead of a char array as returned by `checksum_chars`). If you look at the implementation of `checksum_chars`, they take bytes from a byte array and needlessly transform them to chars.

If rust-bitcoin people start returning non UTF-8 chars as part of their checksum, this will be something to complain about. No one needs emojis in their checksum.

```suggestion
type DescriptorId = [u8; 8];

fn descriptor_id(desc: &Descriptor<DescriptorPublicKey>) -> DescriptorId {
    let mut eng = miniscript::descriptor::checksum::Engine::new();
    eng.input(&desc.to_string()).expect("must!");
    let mut id = DescriptorId::default();
    for (i, c) in eng.checksum_chars().into_iter().enumerate() {
        id[i] = c as u8;
    }
    id
}
```